modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts

```diff
@@ -802,36 +817,52 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       return enterprise.hasFeatureFlags(['useEnterpriseEcdsaTssChallenge']);
     })();
 
+    // create BitGo range proof and paillier proof challenge
+    const createBitgoChallengeResponse = await getTxRequestChallenge(
+      this.bitgo,
+      this.wallet.id(),
+      txRequestId,
+      index.toString(),
+      requestType,
+      modulus.enterprise
+    );
+    const bitgoToEnterprisePaillierChallenge = createBitgoChallengeResponse.p
+      ? { p: createBitgoChallengeResponse.p }
+      : undefined;
+    const enterpriseToBitgoPaillierChallenge = createBitgoChallengeResponse.p
+      ? EcdsaTypes.serializePaillierChallenge({
+          p: await EcdsaPaillierProof.generateP(hexToBigInt(modulus.bitgo)),
+        })
+      : undefined;
+
     if (!shouldUseEnterpriseChallenge) {
       const entChallenge = await EcdsaRangeProof.generateNtilde(minModulusBitLength);
       return {
-        enterpriseChallenge: EcdsaTypes.serializeNtilde(entChallenge),
-        bitgoChallenge: await getTxRequestChallenge(
-          this.bitgo,
-          this.wallet.id(),
-          txRequestId,
-          index.toString(),
-          requestType,
-          'ecdsa'
-        ),
+        enterpriseChallenge: {
+          ...EcdsaTypes.serializeNtilde(entChallenge),
+          p: enterpriseToBitgoPaillierChallenge ? enterpriseToBitgoPaillierChallenge.p : undefined,
+        },
+        bitgoChallenge: {
+          ...createBitgoChallengeResponse,
+          p: bitgoToEnterprisePaillierChallenge ? bitgoToEnterprisePaillierChallenge.p : undefined,
+        },
       };
     }
 
-    const result = await this.wallet.getChallengesForEcdsaSigning();
-    const enterpriseChallenge = result.enterpriseChallenge;
-    const bitgoChallenge = result.bitgoChallenge;
+    // TODO(BG-78764): once the paillier proofs are complete, reduce challenge creation to one API call
+    const walletChallenges = await this.wallet.getChallengesForEcdsaSigning();
```

Comment:
Why are you calling it walletChalleges? Is there anything specific to the challenge per wallet? Thought these were more so just the ent + bitgo challenges.

Reply:
just something different then `result` which was not very descriptive.